src/vm/tests/contracts.rs

```diff
@@ -164,6 +172,89 @@ fn test_get_block_info_eval() {
     }
 }
 
+/// `BurnBlockTestDB` is a mocked out implementation of `BurnStateDB` for test.
```

Comment:
Could we use `TestSimBurnStateDB` here?

Reply:
I mean I guess I'm wondering if another mock of `BurnStateDB` is strictly necessary - if it seems necessary to you, then go ahead with `BurnBlockTestDB`.

Reply:
it's necessary.. we need a mock otherwise we would have to spin up a whole database.. which is a good integration test, but wouldn't be a unit test.

also, we can't use NullBurnStateDB because it always returns None and then we can't do a meaningufl test